recipes/moc/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from os import rename
+from os.path import isdir, join
+from shutil import move
+from glob import glob
+from distutils.version import LooseVersion
+
+class MocConan(ConanFile):
+    name = "moc"
+    homepage = "https://www.github.com/zuut/moc"
+    description = "Moc, the marked-up object compiler"
+    topics = ("conan", "moc", "idl", "code generation")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"    
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = { "fPIC": [True, False] }
+    default_options = { "fPIC": True }
+    _cmake = None
+    _supported_compilers = [
+        ("Linux", "gcc", "6"),
+        ("Linux", "clang", "6")
+    ]
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if not self._check_compiler():
+            raise ConanInvalidConfiguration("%s package is not compatible with os %s and compiler %s version %s." % (self.name, self.settings.os, self.settings.compiler, self.settings.compiler.version))
+
+    def _check_compiler(self):
+        os = self.settings.os
+        compiler = self.settings.compiler
+        compiler_version = tools.Version(compiler.version)
+        return any(os == sc[0] and compiler == sc[1] and compiler_version >= sc[2] for sc in self._supported_compilers)
+
+    def requirements(self):
+        self.requires("flex/2.6.4")
+        self.requires("bison/3.7.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        rename(self.name + "-" + self.version, self._source_subfolder)
```

Comment:
Thanks for the input but I'm wondering exactly how the approval process for getting a package into conan-center is supposed to work. Is this typical?

No doubt your suggestion its simpler but in the guide for creating a package for conan center, https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#how-to-provide-a-good-recipe , they point to crecastnavigation as a good example. this uses rename. Also the fact that there are lots of existing packages already committed to the repo, I don't see this as important to change in order to approve this change. 

Please mark this as approved.

Reply:
This is because our docs are never up to date.... Joy's of being open source we do not have a documentation team :)

Please apply the suggestion as it improves portability for windows users